cider/exec/nextgen/jitlib/base/JITControlFlow.h

```diff
@@ -21,8 +21,32 @@
 #ifndef JITLIB_BASE_JITCONTROLFLOW_H
 #define JITLIB_BASE_JITCONTROLFLOW_H
 
+#include <functional>
+
 #include "exec/nextgen/jitlib/base/JITValue.h"
 
-namespace cider::jitlib {};  // namespace cider::jitlib
+namespace cider::jitlib {  // namespace cider::jitlib
+class IfBuilder {
+ public:
+  virtual ~IfBuilder() = default;
+
+  virtual void build(
```

Comment:
I guess you are try to introduce builder pattern here,  if so, can we separate these method parameters in their own setter method, and then have a build method with empty parameter to build the thing we want.

For example:
 Ifbuilder.condition(xx).if_true(xx).else_block(xx).build();

This library bytecode from airlist does the same thing, for more information see below link:
https://github.com/airlift/bytecode/blob/master/src/main/java/io/airlift/bytecode/control/IfStatement.java#L60
https://github.com/airlift/bytecode/blob/master/src/main/java/io/airlift/bytecode/control/IfStatement.java#L84